persistence/src/main/java/com/epam/esm/dao/GiftCertificateDao.java

```diff
@@ -0,0 +1,72 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.GiftCertificateQueryParameters;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * The interface GiftCertificateDao.
+ *
+ * @author Verbovskiy Sergei
+ * @version 1.0
+ */
+public interface GiftCertificateDao {
+    /**
+     * Add gift certificate
+     *
+     * @param giftCertificate the gift certificate
+     */
+    GiftCertificate add(GiftCertificate giftCertificate);
+
+    /**
+     * Find gift certificate by id.
+     *
+     * @return optional
+     */
+    Optional<GiftCertificate> findById(long certificateId);
```

Comment:
Please add the case to a javadoc when Optional will be returned

Reply:
> Please add the case to a javadoc when Optional will be returned

Thank you, fixed